doc/source/user_guide/deployment_kubernetes.rst

```diff
@@ -95,3 +95,407 @@ Refer `DockerFile <https://github.com/xprobe-inc/xorbits/blob/main/python/xorbit
+
+You should set its maxmemory as 50mb since 2mb in the example is too small.
+
+Make sure redis pod is running:
+
+.. code-block:: bash
+
+    $ kubectl get po redis
+    NAME    READY   STATUS    RESTARTS    AGE
+    redis   1/1     Running   0           6d6h
+
+..
+
+
+Check redis pod's IP address. In this example, IP for redis is 172.17.0.8.
+
+.. code-block:: bash
+
+    $ kubectl get po redis -o custom-columns=IP:.status.podIP --no-headers
+    172.17.0.8
+
+..
+
+You can also check how much CPU and memory resources redis pod gets by running
+
+.. code-block:: bash
+
+    $ kubectl get po redis
+
+..
+
+and check the corresponding fields.
+
+Kubernetes
+~~~~~~~~~~
+Follow previous Kubernetes section to initialize a K8s cluster on your machine.
+
+Install ``kubectl``, a command-line tool for interacting with Kubernetes clusters and verify its installation.
+
+.. code-block:: bash
+
+    $ kubectl version --client
+    WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
+    Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.4", GitCommit:"872a965c6c6526caa949f0c6ac028ef7aff3fb78", GitTreeState:"clean", BuildDate:"2022-11-09T13:36:36Z", GoVersion:"go1.19.3", Compiler:"gc", Platform:"linux/amd64"}
+    Kustomize Version: v4.5.7
+
+..
+
+JuiceFS Installation
+~~~~~~~~~~~~~~~~~~~~
+
+We will still walk you through the process of installing JuiceFS on a Kubernetes cluster, enabling you to leverage its features and benefits.
+
+There are three ways to use JuiceFS on K8S  `Use JuiceFS on Kubernetes <https://juicefs.com/docs/zh/community/how_to_use_on_kubernetes>`_.
+But our implementation in k8s must rely on CSI since CSI provides better portability, enhanced isolation, and more advanced features.
+
+Reference Page: `JuiceFS CSI Driver <https://juicefs.com/docs/csi/getting_started/>`_
+
+
+JuiceFS CSI Driver
+++++++++++++++++++
+
+Installation with Helm
+^^^^^^^^^^^^^^^^^^^^^^
+Reference Page: `JuiceFS Installation with Helm <https://juicefs.com/docs/csi/getting_started#helm-1>`_
+
+1. `Install Helm <https://helm.sh/docs/intro/install/>`_
+
+2. Download the Helm chart for JuiceFS CSI Driver
+
+.. code-block:: bash
+
+    $ helm repo add juicefs https://juicedata.github.io/charts/
+    $ helm repo update
+    $ helm fetch --untar juicefs/juicefs-csi-driver
+
+..
+
+.. code-block:: bash
+
+    $ cd juicefs-csi-driver
+    # Installation configurations is included in values.yaml, review this file and modify to your needs
+    $ cat values.yaml
+
+..
+
+You should be careful with limits and requests of cpu and memory. Change according to your system settings.
+
+.. code-block:: bash
+
+  resources:
+    limits:
+      cpu: 100m
+      memory: 50Mi
+    requests:
+      cpu: 100m
+      memory: 50Mi
```

Comment:
I think what is given here is a minimal configuration that needs to be clarified in the text.